### ISSUE TITLE
Wikimedia Commons Stroke Order project want to use your data

### ISSUE THREAD

**Issue by hugolpz** [skishore/makemeahanzi#25]
Hello there,
I come from the [Stroke Order project](https://commons.wikimedia.org/wiki/Commons:Stroke_Order_Project) and [Ancient Chinese Characters project](https://commons.wikimedia.org/wiki/Commons:Ancient_Chinese_characters_project) on Wikimedia Commons, I just discovered and really love your project and data, which has a much larger reach that our Stroke Order project. 

The SO project has strong UX / design expertise, we discussed a lot about how to display stroke order efficiently for teaching purposes. We come up with [few elegant styles](https://commons.wikimedia.org/w/index.php?search=馬-jbw.png&title=Special:Search&go=Continuer&searchToken=4xufo4zandj34317x7j8dk71z), [naming conventions](https://commons.wikimedia.org/wiki/Template:Chinese_characters_naming) and file formats (300px) : 
![-sbs.gif](https://user-images.githubusercontent.com/1420189/35152545-1523e67c-fd23-11e7-92da-e01237e5efdd.gif)
![-red.png](https://user-images.githubusercontent.com/1420189/35152546-155ad600-fd23-11e7-828e-cb99633d8e3a.png)

I would like to work on your open data to export it as images suitable for Wikimedia projects.

**Comment by parsimonhi** [comment]
Hello,

Here is a demo that generates png images in your 'red' style from svg similar to those of makemeahanzi (the animCJK project is derived from makemeahanzi) : http://gooo.free.fr/animCJK/official/red.php

Hope it helps.

**Comment by hugolpz** [comment]
Thanks Parsimonhi,
I will dig into [this toolchain's licenses](http://gooo.free.fr/animCJK/official/licenses/COPYING.txt) in order to see if i can use these files on Wikimedia Commons, preferably under CC-BY-SA or Public Domain license, as the SO project do.

See also : https://commons.m.wikimedia.org/wiki/Special:MyLanguage/Commons:Copyright_tags

**Comment by parsimonhi** [comment]
@hugolpz 
If you can use Arphic data and makemeahanzi graphic data, you can use animCJK data (the 'red' demo can be easily rewritten using only basic javascript code).

**Comment by skishore** [comment]
Hi Hugo, welcome! I'm familiar with the Stroke Order Project - I looked at that project (and in particular the methods used to create that data) when I started this one. I saw that the data is very high quality, but the approach took a lot of manual work.

Incorporating this dataset into the Stroke Order Project sounds like a good idea, but as you've noted, the main stumbling block is the license mismatch. What are your plans for that?

**Comment by hugolpz** [comment]
As for the various coloring or svg document's design, I'am junior but still can write nodejs scripts myself so I'am sure of the copyrights for the documents design are mine and open license for sure. I would also acquire autonomy.

As for Arphic data : 
1. I could convert the svgs I produce into PNG via imagemagick, so the XML data is not present. When doing so, the shape present in the png is just a common nonsensical representation of Chinese characters and is PD license.
2. I could alter the hand-edit the path so my svg data varies from Arphic. But I'am not super hot for that.

**Comment by parsimonhi** [comment]
@hugolpz
I don't think that you can change the data format (i.e. convert the svgs into png) without mentioning from where you got the data. This is explicitly mentioned in the Arphic license (see https://github.com/skishore/makemeahanzi/tree/master/APL/english).

However, it seems that Wikimedia Commons accepts "freely licensed" works as mentioned at https://commons.m.wikimedia.org/wiki/Commons:Licensing, and according to https://freedomdefined.org/Definition,  it is a "permissible restriction" to have to mention in your files where data come from (i.e. from Arphic data, makemeahanzi data or animCJK data).

**Comment by hugolpz** [comment]
> tl;dr: SO project do not wish to share the Arphic font file nor xml data both under Arphic License, but what to share _prints_ of the fonts, augmented by colors. Prints of fonts are subject to less copyrights, especially when shape is PD due to history. Attaching and require viral conservation of Arphic License and MakeMeAHanzi LGLP license to each image is not practical. AnimCJK will not be used. Polite citation of both Arphic and MakeMeAHanzi makes sense as fair. I will have to ask Commons users with more expertise in Copyright laws and fonts. I'am also looking for alternative data of lower scope (SO project really need ~1000 images).

**Comment by parsimonhi** [comment]
@hugolpz
The issue looks very complicated and I am not an expert anyway.

If you need to use some parts of animCJK project in the futur (especially if you have some interest for Japanese characters which are not part of makemeahanzi), there is no problem.

I hope you will succeed to achieve your project.

**Comment by hugolpz** [comment]
Correction: from a recent talk with Skishore #1, it appears it would be best to colloborate with @parsimonhi's animCJK since you have a dedicated focus on multi-polities support (PRC, ROC, Japan, Korea, etc.).

**Comment by parsimonhi** [comment]
Sample of animated gif for Japanese version of 馬 using animCJK and LICEcap (https://www.cockos.com/licecap/). Same process can be done with Makemeahanzi.
![39340](https://user-images.githubusercontent.com/22865900/35232786-aac60ef0-ff9c-11e7-81a3-d0aef1a17dcc.gif)
One all is in place, it takes less than one minute to record and save a character animation as an animated gif.

**Comment by hugolpz** [comment]
:sob: :100:
Which tech do you use to create your animations ? (Note: I'am rigth now looking into your repos) If nodejs I can come and hack it further for minor things, like adding a `stroke:2px solid black;` or `stroke="black" stroke-width="2"`

**Comment by hugolpz** [comment]
This is out of my reach as of now. Already too many projects in my life.
I will focus on clarifying the licenses and attacking the -red.png via nodejs, that I know a bit.

**Comment by parsimonhi** [comment]
Finally, I dit it (pure javascript+HTML, nothing else):
http://gooo.free.fr/animCJK/official/samples/imageFactory.html
Note 1: the svgs folder of the demo online contains only the MakeMeAHanzi svg for the characters 鼠貓牛虎兔龍蛇馬羊猴雞狗
Note 2: the same code can generate images for any characters of animCJK project or MakeMeAHanzi project. Just put the svg files you need in the svgs folder (see below)

To make a demo on your own website that use all the svgs of MakeMeAHanzi project:

1) create a folder named "imageFactory" (any other name is ok)

2) create a subfolder of your "imageFactory" folder called "samples"

3) get "imageFactory.html" file from animCJK project
(https://github.com/parsimonhi/animCJK/tree/master/samples),
put it in your "samples" folder.

4) create a subfolder of your "samples" folder called "_js"

5) get "Animated_GIF.js" and "magicAcjk.js" from animCJK project
(https://github.com/parsimonhi/animCJK/tree/master/samples/_js),
put them in your "_js" folder.
Note that "Animated_GIF.js" can also be downloaded from Animated_GIF project
(https://github.com/sole/Animated_GIF/dist).
Note that Animated_GIF project itself is derived from several other projects.
Update 2018/11/05: get also "brushingAcjk.js" from animCJK project.

6) get the "svgs" folder from MakeMeAHanzi project
(https://github.com/skishore/makemeahanzi),
put it in your "imageFactory" folder,
or if you already put the "svgs" folder elsewhere on your website,
replace "../svgs/" in "imageFactory.html" file by a relative path to your "svgs" folder.

You can also use the svg of animCJK that are in its "svgsJa" and "svgsZhHans" folders
(https://github.com/parsimonhi/animCJK/tree/master),
but here, it's the MakeMeAHanzi project blog! ;-)

Pending issue: no transparency for the animated gif images.

Any comments are welcome.

Have fun!

**Comment by hugolpz** [comment]
1) **@parsimonhi : are you French**........................................................... ?
About 80% of CJK opensources is made by French dudes. Kanjivg, Wikimedia CJK project, AnkiDroid, Chinese audios via LinguaLibre.org for HSK (ex "Shtooka Recorder") and Tatoeba.org for sentenses were initiated by French nerds. For some reasons French love this field.

2) **Language:**  If I get it right, your demo `*-red.png`-like and `*-sbs.gif`-like are generated via js right ? I matters because I can hack js well while I have zero php experience. Minor edits may help, such as for file names :D Or I may provide you with naming conventions and guidelines for code / design improvement request ?

EDIT:

3) Ok, below is my **review** with Wikimedia Commons CJK guidelines in mind :

- :heavy_check_mark::grey_question: 'Transparent PNG' : your app's new strokes are printed UPON. Commons print them UNDER. but well... This also works.
  -  Recommended file format : `*-red.png`
- :x:  'Not transparent PNG' : not needed. ImageMagick allows mass conversion to jpg if needed.
- :heavy_check_mark: 'Animated GIF with grey strokes' : perfect to me !
  -  Recommended file format : `*-sbs.gif`
- :x: 'Animated GIF without grey strokes' : not needed. If learning is the objective, then we need the grey background.

4) **License :** I encourage you to publish under `CC-0 {your name or pseudo}`. It's a Public domain, yes. Wikimedia CJK Stroke Order project contributors including myself decided to contribute under such license on the basis that Chinese character and calligraphy are themselves Public Domain and part of the [intangible cultural heritage](https://en.wikipedia.org/wiki/UNESCO_Intangible_Cultural_Heritage_Lists). With CC-0 / Public Domain license, these files can be publish on Commons, and link back to your project and/or author page can still be indicated out of fairness. I would recommend you to use and add such license on your page.

**Comment by parsimonhi** [comment]
1) I am french!
2) I have now scripts that generate  all the images automatically (need more than one hour to generate all the images: i am trying to reduce this time). Can you please open an issue on https://github.com/parsimonhi/animCJK/issues in order to discuss about how they work, how they can be adapted if necessary, and what storage rules have to be applied?
3) According to me, new strokes have to be printed UPON of course. Never see a calligrapher writing new strokes under previous strokes! I noticed that some (but not all) "red" images of Wikimedia commons have the new strokes writing under the previous ones. This is at least disturbing and probably an error. Il you want red strokes under, it should be better to use red for the first stroke to black for the last stroke.
4) Ok for a CC license. As for any image that contains some texts, it makes sense that the license concerning the font used for drawing texts is not "transmitted" to the license of the image.
Update 2018/11/08: finally, after reading many documents about font licensing, I thinks that distributing images derivated from Arphic fonts under a CC license is questionable. Arphic license seems required.

**Comment by hugolpz** [comment]
EDIT : [holy f%$# of awesomeness](http://gooo.free.fr/animCJK/official/samples/imageFactory.html
) ! [I'am nerd-crying](https://github.com/skishore/makemeahanzi/issues/25#issuecomment-432469280) 😭  How did I miss that ?
Important: On commons we tested transparent background gif but it was not doing well, maybe due to imagemagick, so your image factory is finished.@parsimonhi I'am soooo sooorrrrryyy !!!!!!!  😭   😭   😭   😭  

I'am resuming [efforts](https://hugolpz.github.io/animCJK/) on this side and I now have a wikibot ! : )

**Comment by hugolpz** [comment]
Edit. @parsimonhi ❤️

**Comment by hugolpz** [comment]
Wikimedia Commons [Template:PD-font](https://commons.m.wikimedia.org/wiki/Template:PD-font)
![Screenshot_2021-03-29-16-35-13-224](https://user-images.githubusercontent.com/1420189/112853243-f25eb600-90ac-11eb-9dbc-32dca9131a19.jpeg)
